### ISSUE TITLE
Add to f-droid

### ISSUE THREAD

**Issue by onodera-punpun** [mpv-android/mpv-android#37]
It's a store just for free software and doesn't depend on GAPPS.

**Comment by 1C3** [comment]
I'm going to try to add it soon

**Comment by 1C3** [comment]
Here is the opened [merge request](https://gitlab.com/fdroid/fdroiddata/merge_requests/2124), if the f-droid staff is quick to answer the app should be on the main f-droid repo in a couple of days

**Comment by Floens** [comment]
This doesn't compile the native libs, does it?

**Comment by 1C3** [comment]
Whoops I didn't even think about them, and to be honest now that I checked I don't really know how to include them in the build process right now. I'll look it up over the weekend if I have time but can't promise anything.

**Comment by Floens** [comment]
They've merged the broken build, and I've given them a heads-up about it.

**Comment by relan** [comment]
mpv-android's build system does not fit well into F-Droid's workflow.

`buildscripts` bootstraps `mpv-android` and its dependencies. For F-Droid, an app repo is a central point that references dependencies. This is a common case in Android world.

There are other issues with `buildscripts`: it uses `sudo`, installs packages via `yum` or `apt-get`, downloads SDK and NDK, creates toolchain installations, downloads tarballs with libraries source code, pulls HEAD revisions of some libraries. All those things may not be done on F-Droid build server.

Usually apps that need native libraries reference them as Git submodules and compile them during the whole project building, see https://developer.android.com/studio/projects/add-native-code.html. This would play much better with F-Droid.

**Comment by ShapeShifter499** [comment]
This could be added to the IzzySoft F-Droid repository in the interim.

/cc @izzysoft

**Comment by IzzySoft** [comment]
Yes, can be done. As my repo runs on "private space", there's a size limit I've set up, reserving ~20M per app to get the most out of it. With this `.apk` being about 12M, that means I can keep 1 version (always the latest, of course). Updates as usual, within 24h of the `.apk` being attached to a new tag/release.

Will see to it as soon as I'm back at my "work bench". Thanks for the suggestion and for pinging me, @ShapeShifter499 – no idea why this repo slipped through my net until now. Most likely because it has no `README.md` on its main page, explaining what the repo is about – and the term "MPV" could mean everything (like Material Profile Viewer, My Private Vault, Master Pr0n Vehicle… oops…).

So far I gather it's a wrapper around some library (libmpv). Can any one tell me if it's an end-user application, or rather a developer tool? Not even the playstore description is giving a clue:

> A simple Android wrapper around the libmpv library provided by mpv. Android developers welcome to contribute!

So is it a video player for end users, or just a "better library" for devs? A more detailed description I'd need in order to add it to the repo ;)

**Comment by sfan5** [comment]
It's an actual video player for end users, though as the description implies it's a rather thin wrapper around libmpv without lots of extra features.

**Comment by IzzySoft** [comment]
Thanks, @sfan5 – so may I suggest to alter the "short desc" (summary) of the project from

> #mpv-android @ freenode

To

> Small and light-weight video player for Android; #mpv-android @ freenode

to at least give an idea what the project is about – and also add at least a rudimentary `README.md`? If your users already approach me to have it in a repo your app must be "somewhat good" (confirmed by the Playstore ratings), and it'd be a pity if others miss it just because there's no description (which was why I initially skipped it). Playstore at least gives a little idea by the screenshots, though the desc there could need a few more details on what to expect as well. :wink:

I'm just suggesting, not complaining!

**Comment by IzzySoft** [comment]
OK, find it [here](https://apt.izzysoft.de/fdroid/index/apk/is.xyz.mpv) now. Enjoy!

**Comment by IzzySoft** [comment]
PS, just a remark: due to the tag naming not reflecting the version number, I currently have to check for updates in the manual run (which I do about twice a month – as opposed to the automated daily check). Be welcome to drop me a note if you switch to a different tag-naming scheme (most projects use a format like `v1.2.3` which works perfectly; `2017-06-14_1.2.3` the updater could deal with as well (in that order – so `v1.2.3_2017-06-14` would not work). As yours is the only project so far with the currently used naming schema, I didn't feel like implementing a separate check-algorithm for that, hope you understand :wink:

**Comment by radamar** [comment]
+1

**Comment by c-e-p-x-u-o** [comment]
https://f-droid.org/packages/is.xyz.mpv/

404 Page Not Found

Please reopen.

**Comment by IzzySoft** [comment]
@c-e-p-x-u-o https://apt.izzysoft.de/fdroid/index/apk/is.xyz.mpv (it is in my repo). Though it probably should meet the inclusion criteria of the official repo from what I see. Feel free to open an RFP -- if @sfan5 approves 😉

**Comment by KaKi87** [comment]
Hello,
I can't find mpv in F-Droid ?
Thanks

**Comment by IzzySoft** [comment]
@KaKi87 read the comment just above yours. You'd first need to add my repo to your client – or with the browser, use the URL mentioned. F-Droid is not a single centralized "store", there are many (3rd-party) repos available as well :wink:

**Comment by KaKi87** [comment]
Well, I'm pretty sure the original request was about adding mpv to the centralized store.

Having extra steps is anti-UX and manually downloading the APK via GitHub releases becomes even easier.

Thanks

**Comment by IzzySoft** [comment]
I'm pretty sure you're correct. But at F-Droid.org, the app has only a single (old) build which in addition is disabled, auto-update being disabled as well – which is why you cannot find it there. As pointed out above, current versionName/tagName pair is a bit troublesome and cannot be matched automatically: `versionName: "2020-10-13-release"` vs tag name "mpv-android-2020-10-13" doesn't match. We could match by tag `mpv-android-%v` – but for that, `versionName` must come without that `-release` suffix.

**TL;DR:** As long as the two do not match, F-Droid cannot automatically detect and build updates. So each new version you want there would need its own MR to be created and processed. Apart from not being really practicable and (needlessly) taking a lot of time on both ends, nobody opened such MRs yet.

Until that's done, I guess adding another repo (with currently close to 600 additional apps) is the better UIX compared to sending the users here to grab the APK manually from releases. Though I agree, it would be even better were it available directly at F-Droid.org. Once above mismatch is solved, we can give that a try.

**Comment by KaKi87** [comment]
> versionName must come without that `-release` suffix

Why not just removing it then ?

> adding another repo (with currently close to 600 additional apps) is the better UIX

Have these additional apps passed the same inclusion policy as those applied to the centralized store ?

**Comment by IzzySoft** [comment]
> Why not just removing it then ?

I can't, but that would indeed help. @xyzz / @sfan5 ?

> Have these additional apps passed the same inclusion policy as those applied to the centralized store ?

Not the same, but comparable. Rules in my repo are a little more relaxed; consider it as something between universe and nonfree for Debian. Find the [details here](https://gitlab.com/IzzyOnDroid/repo).

**Comment by KaKi87** [comment]
> most common examples include „blobs“ or dependencies on proprietary frameworks

Well, users should expect all apps from F-Droid to not have apps that include these.
FLOSS is FLOSS.

**Comment by IzzySoft** [comment]
Agreed. But I didn't yet hear from any adult who was born fully grown. As a German proverb goes: "Es ist noch kein Meister vom Himmel gefallen." (literally: "No master has yet fallen from heaven."). About 200 apps were able to finally move over to F-Droid.org from my repo once they solved those issues. Until then, they are at least available outside Google's walled garden. I do not blame people for things they were not aware of when beginning to write their app – but encourage and help (if possible) to get rid of "unwanted stuff" and finally "move over".

**TL;DR:** I agree, but there's a road to walk to reach the goal.

Now let's hope devs here will solve the (comparably simple) task of "cleaning up" the `versionName` – and I can help then (re)activating the app at F-Droid (disclosure: I'm one of the maintainers there).

PS: There are "apps from F-Droid(.org)", and they fullfil your requirements. And there are apps from "3rd party repos using the F-Droid (backend) software" – and they can have their own rules each. Don't mix the software with the organization(s) using it :wink:

**Comment by linsui** [comment]
I open a [mr](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/8298) in F-Droid side. Please take a look. Could you please add fastlane here?

**Comment by sfan5** [comment]
I have added fastlane now.

Regarding your f-droid MR: when you build mpv-android you should really build the native code for `arm64` too.

**Comment by linsui** [comment]
Thanks! For test purpose I only build the arm arch but it's pretty easy to add arm64 support, thanks to the build script.

**Comment by sfan5** [comment]
@linsui If it helps there is now a [tagged release](https://github.com/mpv-android/mpv-android/tree/mpv-android-2021-03-10) that includes the fastlane metadata I added last month.

**Comment by linsui** [comment]
Thanks! Yeah, I'm waiting for that and I'll update my MR soon.

**Comment by linsui** [comment]
@sfan5 I didn't notice that the fastlane structure is not correct... It should be `fastlane/metadata/android`. Tha `android` is missing so F-Droid can fetch the metadata. Could you please fix it and move the tag? Thanks!

**Comment by sfan5** [comment]
I will not retroactively change the tag but you can use a new one: [`mpv-android-2021-03-10-fastlane`](https://github.com/mpv-android/mpv-android/releases/tag/mpv-android-2021-03-10-fastlane)

**Comment by linsui** [comment]
Thanks!

**Comment by linsui** [comment]
Finally! https://f-droid.org/en/packages/is.xyz.mpv/ 🎉 @sfan5 You can put [the badge](https://gitlab.com/fdroid/artwork/-/blob/master/badge/README.md) in the Readme.

[<img src="https://fdroid.gitlab.io/artwork/badge/get-it-on.png"
    alt="Get it on F-Droid"
    height="80">](https://f-droid.org/en/packages/is.xyz.mpv/)

**Comment by 532910** [comment]
Update https://github.com/mpv-android/mpv-android/blob/master/README.md please!

**Comment by IzzySoft** [comment]
And if there're no objections, I'd remove the app from my repo in about a week (see [FAQ](https://gitlab.com/IzzyOnDroid/repo/-/wikis/FAQ)). If you want it to stay, just say.

**Comment by sfan5** [comment]
Thanks for bringing mpv-android to f-droid, I have updated the README badge.

@IzzySoft No objects to removing it from your repo, it's not needed anymore after all.

**Comment by IzzySoft** [comment]
Thanks! Marking it then. Should at any point you need it back one day, just yell :smile: I keep a backup of the metadata, so re-instantiating would be pretty easy.
